virtual_rainforest/core/model.py

```diff
@@ -1,12 +1,42 @@
-"""The `core.model` module.
-
-The `core.model` module defines the api that all individual models (e.g. the soil
-model) should conform to. This consists of a class (`Model`), which defines the expected
-functions. The relevant modules will create classes to represent specific models, which
-will inherit from the `Model` base class. These subclasses will generally overwrite the
-functions defined in the base class, which are defined mainly to force a consistent api
-between models. It also establishes a model registry that allows models to become
-accessible across scripts without individual loading in.
+"""API documentation for the :mod:`core.model` module.
+************************************************** # noqa: D205
+
+The `core.model` module defines the api that all individual models (e.g. the soil model)
+should conform to. This consists of a class (`BaseModel`), which defines the expected
+functions. This class is an abstract base class, which is a class that is never intended
+to be instantiated itself but instead serves as a blueprint for inheriting classes. Some
+of its functions will be inherited and used by its child classes (e.g. `__repr__` and
+`__str__`), unless they are explicitly overwritten in the child class. However, it also
+possesses abstract methods (denoted by `@abstractmethod`), which are merely
+placeholders. For these abstract methods child classes have to overwrite the methods
```

Comment:
```suggestion
placeholders. For these abstract methods, child classes have to overwrite the methods
```
This reads a bit easier with a comma  :-)

Reply:
Also, could you give an example for a case when a child overwrites an abstract method?